### ISSUE TITLE
Worse results with pytorch 1.0

### ISSUE THREAD

**Issue by panji1990** [ClementPinard/SfmLearner-Pytorch#54]
I cloned the latest code and trained the model from scratch on KITTI raw. Here is depth result using Eigen split on KITTI:

|Abs Rel | Sq Rel | RMSE | RMSE(log) | Acc.1 | Acc.2 | Acc.3|
|-- | -- | -- | -- | -- | -- | --|
|0.2260 | 1.8739 | 6.4835 | 0.3061 | 0.623 | 0.880 | 0.951|

which is worse than the results you reported. Has anyone encountered the same issue?

**Comment by panji1990** [comment]
Hi Clement,
It seems that it's because some static/slow-motion frames are not removed. Can you please share your sequence folder names after pre-processing? Thanks

**Comment by ClementPinard** [comment]
Hello, This maybe due to train/val split that did not take into account the camera number before (so there was the possibility of sequence camera_2 in train and camera_3 in val.

I'm not gonna lie, I struggled to get these results in the first place, with a very specific set of hyperparameters, probably dependant on the implementation itself. The reason for this is that using the exact same parameters and algorithm as tensorflow original implementation did not make the network converge at all !

I don't have much time right now, so I can't investigate the "new ideal set of hyperparameters and smoothing losses" right now, but what you can do is :

 * test the algorithm with the stacked dataset, possibly constructed from original code [here](https://github.com/tinghuiz/SfMLearner/tree/master/data). In case there was a regression in constructing the dataset.
 * Try to apply smooth loss on disparity, the same way it was done before. Be careful, a smooth loss of 0.5 and loss downscale of 2 per depth scale (which is the hyperparameters used by original paper) does not work ! but you can tries values around. I recall a smooth loss on disparity of 0.5 and loss downscale of ~2.5 got good results.
 * Try to train on the whole dataset and not just the train set. Original implementation did not use validation set and just trained the network and tested it after different number of iterations.
